### PR TITLE
Isolate citation updates into their own PRs

### PR DESCRIPTION
**Pull request by chartgerink**
This PR isolates the action to update the `CITATION.cff`. It is run whenever the relevant files change on the main branch, and creates a pull request if anything changed. 

This was previously discussed in #128 - I tested it as well on a [test repo](https://github.com/chartgerink/testpkg/actions/runs/8077066099).

### CODE REVIEW

File: .github/workflows/update-citation-cff.yaml [hunk 1]
```diff
@@ -8,21 +8,17 @@ on:
     branches:
       # This may seem like a no-op but it prevents triggering on tags.
       # We use '**' rather '*' to accomodate names like 'dev/branch-1'
```
Comment (Bisaloo) [inline .github/workflows/update-citation-cff.yaml:10]: This now outdated comment should be removed

**Review by Bisaloo — APPROVED** [review]

File: .github/workflows/update-citation-cff.yaml (hunk 1 above)
Comment (chartgerink) [inline .github/workflows/update-citation-cff.yaml:10]:
```suggestion
```

Comment (chartgerink) [reply .github/workflows/update-citation-cff.yaml:10]:
```suggestion
```

**Comment by Bisaloo** [comment]
Thanks!

**Pull request merged**